example_async.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+"""
+This is an example of how the pytradfri-library can be used async.
+
+To run the script, do the following:
+$ pip3 install pytradfri
+$ Download this file (example_sync.py)
+$ python3 test_pytradfri.py <IP> <KEY>
+
+Where <IP> is the address to your IKEA gateway and
+<KEY> is found on the back of your IKEA gateway.
+"""
+
+import asyncio
+import logging
+import sys
+
+import pytradfri
+
+root = logging.getLogger()
+root.setLevel(logging.INFO)
+
+try:
+    # pylint: disable=ungrouped-imports
+    from asyncio import ensure_future
+except ImportError:
+    # Python 3.4.3 and earlier has this as async
+    # pylint: disable=unused-import
+    from asyncio import async
+    ensure_future = async
+
+
+@asyncio.coroutine
+def observe(api, device):
+    def callback(updated_device):
+        light = updated_device.light_control.lights[0]
+        print("Received message for: %s" % light)
+
+    observe_command = device.observe(callback, duration=120)
+    task = api(observe_command)
+    ensure_future(task)
+    print('Sleeping to start observation task')
+    yield from asyncio.sleep(1)
+
+
+@asyncio.coroutine
+def run():
+    # Assign configuration variables.
+    # The configuration check takes care they are present.
+    api = pytradfri.aio_api_factory(sys.argv[1], sys.argv[2])
+
+    gateway = pytradfri.gateway.Gateway()
+
+    devices_command = gateway.get_devices()
+    yield from api(devices_command)
+    devices_commands = devices_command.result
+    devices = []
+    for device_command in devices_commands:
+        yield from api(device_command)
+        devices.append(device_command.result)
+
+    lights = [dev for dev in devices if dev.has_light_control]
+
+    tasks_command = gateway.get_smart_tasks()
+    yield from api(tasks_command)
```

Comment:
yielding from the api will actually return the result, no need to fetch it from the command afterwards.